File: Mooga/src/de/lumpn/mooga/ranking/impl/ScoreComparator.java
package de.lumpn.mooga.ranking.impl;

import java.util.Comparator;

/**
 * Compares individuals by score of specific attribute.
 */
public final class ScoreComparator implements Comparator<CrowdingDistanceIndividual> {

	public ScoreComparator(int attribute) {
		this.attribute = attribute;
	}

	@Override
	public int compare(CrowdingDistanceIndividual a, CrowdingDistanceIndividual b) {
		double score1 = a.getIndividual().getScore(attribute);
		double score2 = b.getIndividual().getScore(attribute);
		return Double.compare(score1, score2);
	}

	private final int attribute;
}

File: csharp/InfiniteZelda/Mooga/ScoreComparator.cs
using System.Collections.Generic;

namespace Lumpn.Mooga
{
    /// Compares individuals by score of specific attribute.
    public sealed class ScoreComparator : IComparer<CrowdingDistanceIndividual>
    {
        private static readonly Comparer<double> comparer = Comparer<double>.Default;

        public ScoreComparator(int attribute)
        {
            this.attribute = attribute;
        }

        public int Compare(CrowdingDistanceIndividual a, CrowdingDistanceIndividual b)
        {
            double score1 = a.Individual.getScore(attribute);
            double score2 = b.Individual.getScore(attribute);
            return comparer.Compare(score1, score2);
        }

        private readonly int attribute;
    }
}